src/dotnet/MulticoreJitActivator.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;    
+using System.Runtime.Loader;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.Common;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class MulticoreJitActivator
+    {
+        public bool TryActivateMulticoreJit()
+        {
+            var disableMulticoreJit = IsMulticoreJitDisabled();
+                
+            if (disableMulticoreJit)
+            {
+                return false;
+            }
+
+            StartCliProfileOptimization();
+            
+            return true;
+        }
+        
+        private bool IsMulticoreJitDisabled()
+        {
+            return Environment.GetEnvironmentVariable("DOTNET_DISABLE_MULTICOREJIT") == "1";
```

Comment:
We have a helper for this `Env.GetEnvironmentVariableAsBool` the advantage is the consistent handling of "true" and "false" boolean environment variables:

```
            switch (str.ToLowerInvariant())
            {
                case "true":
                case "1":
                case "yes":
                    return true;
                case "false":
                case "0":
                case "no":
                    return false;
                default:
                    return defaultValue;
            }
```